source/common/http/conn_manager_impl.cc

```diff
@@ -341,7 +345,7 @@ void ConnectionManagerImpl::ActiveStream::chargeTracingStats(
     connection_manager_.config_.tracingStats().client_enabled_.inc();
     break;
   case Tracing::Reason::HealthCheck:
-    connection_manager_.config_.tracingStats().health_check_.inc();
+    // At this point we have not run any filters, including HC one, so this should not happen.
```

Comment:
then take this case statement out, add a default, and throw an invalid argument exception to crash (and add test to that effect).